File: src/main/java/no/fintlabs/mapping/SkjermingMappingService.java
package no.fintlabs.mapping;

import no.fint.model.resource.arkiv.noark.SkjermingResource;
import no.fintlabs.model.instance.SkjermingDto;
import org.springframework.stereotype.Service;

@Service
public class SkjermingMappingService {

    public SkjermingResource toSkjermingResource(SkjermingDto skjermingDto) {
        if (skjermingDto == null) {
            return null;
        }
        SkjermingResource skjermingResource = new SkjermingResource();
        skjermingResource.addTilgangsrestriksjon(skjermingDto.getTilgangsrestriksjon());
        skjermingResource.addSkjermingshjemmel(skjermingDto.getSkjermingshjemmel());
        return skjermingResource;
    }

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/dispatch/mapping/SkjermingMappingService.kt
package no.novari.flyt.archive.gateway.dispatch.mapping

import no.novari.fint.model.resource.Link
import no.novari.fint.model.resource.arkiv.noark.SkjermingResource
import no.novari.flyt.archive.gateway.dispatch.model.instance.SkjermingDto
import org.springframework.stereotype.Service

@Service
class SkjermingMappingService {
    fun toSkjermingResource(skjermingDto: SkjermingDto?): SkjermingResource? =
        skjermingDto?.let {
            SkjermingResource().apply {
                skjermingDto.tilgangsrestriksjon?.let(Link::with)?.let(::addTilgangsrestriksjon)
                skjermingDto.skjermingshjemmel?.let(Link::with)?.let(::addSkjermingshjemmel)
            }
        }
}
